WalletWasabi/Stores/IndexStore.cs

```diff
@@ -88,10 +91,34 @@ public async Task InitializeAsync(CancellationToken cancellationToken)
 				await Task.Run(MigrateToSqliteNoLock, cancellationToken).ConfigureAwait(false);
 			}
 
+			// If the automatic migration to SQLite is stopped or somehow disrupted, we would not delete the old index data.
+			// So check it every time.
+			RemoveOldIndexFilesIfExist();
+
 			await InitializeFiltersNoLockAsync(cancellationToken).ConfigureAwait(false);
 		}
 	}
 
+	private void RemoveOldIndexFilesIfExist()
+	{
+		if (File.Exists(OldIndexFilePath))
+		{
+			try
+			{
+				File.Delete(OldIndexFilePath);
+				File.Delete($"{OldIndexFilePath}.dig"); // No exception is thrown if file does not exist.
+				File.Delete(OldImmatureIndexFilePath);
+				File.Delete($"{OldImmatureIndexFilePath}.dig");
```

Comment:
I would reverse the order of file removals, so `OldIndexFilePath` will be the last one. In that case there must be no leftover if `OldIndexFilePath` gets removed, but others don't.